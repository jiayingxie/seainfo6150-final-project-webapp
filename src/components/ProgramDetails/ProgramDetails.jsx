import React, { Component } from 'react';
import { Link } from "react-router-dom";
import styles from './ProgramDetails.module.css';

export default class ProgramDetails extends Component {
    render() {
        const { department } = this.props;
        const { programName, programOverview } = this.props.program;
        return (
            <div className={styles.container}>
                <Link to={`/engineering/${department.departmentID}`} className={styles.back}>Back to {department.departmentID} page</Link>
                <h1>{programName} </h1>
                <h2>{department.departmentID} department</h2>
                <div className={styles.text} dangerouslySetInnerHTML={{ __html: programOverview }} />
            </div>
        )
    }
}
