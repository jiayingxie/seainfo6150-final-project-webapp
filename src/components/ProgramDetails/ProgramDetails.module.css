.container {
    display: flex;
    flex-flow: column wrap;
    align-items: center;

    border: 1px dashed #666;
    border-radius: 10px;
    box-shadow: 0px 0px 10px 5px rgba(102,169,201, 0.8);
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 15px;
}

.back{
    align-self: flex-end;
    padding: 5px;
    background-color: wheat;
    border-radius: 10px;
    color: black;
}

h1 {
    justify-self: center;
    font-family: cursive, sans-serif;
    font-size: 28px;
}

h2 {
    font-family: Georgia, Times, sans-serif;
    font-size: 24px;
}

.text {
    font-size: 16px;
    columns: 1 200px;
}

@media (min-width: 600px) {
    h1 {
        font-size: 32px;
    }
    
    h2 {
        font-size: 28px;
    }

    .text {
        font-size: 18px;
        columns: 2 200px;
    }
}
  
@media (min-width: 900px) {
    h1 {
        font-size: 36px;
    }
    
    h2 {
        font-size: 32px;
    }

    .text {
        font-size: 20px;
        columns: 3 200px;
    }
}