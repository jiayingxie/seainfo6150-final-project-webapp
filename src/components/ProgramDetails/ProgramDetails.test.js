import React from "react";
import ProgramDetails from "./ProgramDetails";

describe("ProgramDetails tests", () => {
    let department;
    let program;
    // the method is to get test data
    const fetchDepartment = async () => {
        const mul = "multidisciplinary";
        const CSECS = "CSECS";
        const response = await fetch(`https://demo2247515.mockable.io/engineering`);
        const responseJson = await response.json();
        department = await responseJson[mul];
        program = await Object.values(department.programs).find(prog => { return prog.programID === CSECS });
        return program;
    };
    it("renders correctly when passed a program prop", async () => {
        const programData = await fetchDepartment();
        const { container } = await render(<ProgramDetails department={department} program={programData} />);
        expect(container).toMatchSnapshot();
    });
});