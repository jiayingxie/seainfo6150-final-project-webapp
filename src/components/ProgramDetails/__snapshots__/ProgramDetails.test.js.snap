// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramDetails tests renders correctly when passed a program prop 1`] = `
<div>
  <div
    class="container"
  >
    <a
      class="back"
      href="/engineering/multidisciplinary"
    >
      Back to 
      multidisciplinary
       page
    </a>
    <h1>
      Computer Systems Engineering with Concentration in Software Design Engineering
       
    </h1>
    <h2>
      multidisciplinary
       department
    </h2>
    <div
      class="text"
    >
      <p>
        Our computer systems engineering program takes a sociotechnical, engineering approach to software. This engineering foundation is designed to enable students to embrace real-world complexity as a golden opportunity, especially for the more technically advanced student. We are committed to shaping our students to be intuitive problem solvers, experienced engineering architects, and result leaders who will have a great impact at the exciting three-way intersection of computer science, engineering, and ethics.
      </p>
      Our program offers a multitude of courses in big-data engineering and analytics in addition to supplementary courses that are required to deliver the data-analytics results in a meaningful way to management. We cover data management, advanced data management, business intelligence, column databases, data science, and big-data engineering. We offer advanced functional programming using the powerful Scala language and a course on advanced data science as well as cloud computing. Multi-thread concurrent computing is also offered as it is important for synchronizing a huge set of servers working in parallel to do large-scale analytics to make things run faster by a hundredfold increase in speed. Due to the high-level mathematical operations required to run these programs, only software engineers have the capacity to work in such complicated areas. Only they can make the necessary mathematical algorithms execute quickly enough to get the finest results.
      <p>
        Our engineers become fluent in data science for the sake of building the actual system. They study how to write machine-learning algorithms on top of statistical packages.
      </p>
      <p>
        Students may complete a Master of Science in Computer Systems Engineering with Concentration in Software Design Engineering in addition to earning a Graduate Certificate in Engineering Leadership. Students must apply and be admitted to the Gordon Engineering Leadership Program in order to pursue this option. The program requires fulfillment of the 16-semester-hour curriculum required to earn the Graduate Certificate in Engineering Leadership, which includes an industry-based challenge project with multiple mentors. The integrated 32-semester-hour degree and certificate will require 16 hours of advisor-approved software design engineering technical courses.
      </p>
    </div>
  </div>
</div>
`;
